SU2_CFD/src/fluid/CCoolProp.cpp

```diff
@@ -55,29 +55,71 @@ void CCoolProp::SetTDState_rhoe(su2double rho, su2double e) {
   dPde_rho = fluid_entity->first_partial_deriv(CoolProp::iP, CoolProp::iUmass, CoolProp::iDmass);
   dTdrho_e = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iDmass, CoolProp::iUmass);
   dTde_rho = fluid_entity->first_partial_deriv(CoolProp::iT, CoolProp::iUmass, CoolProp::iDmass);
-  if (fluid_entity->phase() == 6) {
+  if (fluid_entity->phase() == CoolProp::iphase_twophase) {
+    // assume it is pure gas
+    fluid_entity->specify_phase(CoolProp::iphase_gas);
+    if (abs(Pressure / Pressure_Critical - 1) < epsilon) {
```

Comment:
Please put this modification of the pressure in a private function of the class so that you dont need to duplicate the code.

Reply:
Thanks for your advice.

I added **check_pressure(P)** as a private function in **CCoolProp.h** and reuse it in **CCoolProp.cpp**. 

The code is more concise.